RestaurantReservation.Db/Services/EntityService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using AutoMapper;
+using FluentResults;
+using RestaurantReservation.Db.Repositories.IRepositories;
+using RestaurantReservation.Db.Services.IServices;
+
+namespace RestaurantReservation.Db.Services
+{
+    public class EntityService<TEntity, TDto, TIEntityRepository> : IEntityService<TDto> where TEntity : class where TDto : class where TIEntityRepository : IEntityRepository<TEntity>
+    {
+        protected readonly TIEntityRepository _entityRepository;
+        protected readonly IMapper _mapper;
+
+        public EntityService(TIEntityRepository entityRepository, IMapper mapper)
+        {
+            _entityRepository = entityRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<Result> CreateAsync(TDto dto)
+        {
+            try
+            {
+                var entity = MapDtoToEntity(dto);
+                await _entityRepository.CreateAsync(entity);
```

Comment:
sometimes it's better to return the created entity (with the Id populated) or the Id at least.